eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -65,13 +65,16 @@ jobs:
         CMAKE_GENERATOR: 'Visual Studio 16 2019'
         CMAKE_GENERATOR_PLATFORM: x64
         BuildArgs: '--parallel 8'
-      MacOS_x64:
```

Comment:
Don't we still want macOS ?

Reply:
We are keeping MacOS testing here: https://github.com/Azure/azure-sdk-for-cpp/blob/817e538db98c0c0005d2d7f267f0a3ca60fc46c5/eng/pipelines/templates/jobs/archetype-sdk-client.yml#L123

But yes, this eliminates one mac build but keeps the mac unit testing build. I took from our discussion that this was ok. If not I can add the mac Mac build-only job back in.